### PR TITLE
Add new binarytrees version

### PR DESCRIPTION
**Pull request by jabraham17**
Adds a new version of the binarytrees benchmark. This is the same as `binarytrees3`, except instead of making the outer loop parallel this makes the inner loop parallel. 

Perf testing on a 4-core internal system
- binarytrees3: 8.6 sec
- binarytrees-inner: 7.2 sec

[Reviewed by @jeremiah-corrado]

### CODE REVIEW

File: test/studies/shootout/binary-trees/binarytrees-inner.good
Comment (jeremiah-corrado) [inline test/studies/shootout/binary-trees/binarytrees-inner.good:1]: does this test produce the same output as the other binarytrees test(s), and can it share a good file?

**Review by jeremiah-corrado — APPROVED** [review]
Looks good! Just one minor suggestion:

File: test/studies/shootout/binary-trees/binarytrees-inner.good
Comment (jabraham17) [reply test/studies/shootout/binary-trees/binarytrees-inner.good:1]: It has the same output as all of the other binarytrees and could share a .good file, but none of the other binarytree variants do this. They all have their own .good file which is duplicated

Comment (jeremiah-corrado) [reply test/studies/shootout/binary-trees/binarytrees-inner.good:1]: Okay makes sense. I'd agree with keeping it as is then.

**Comment by bradcray** [comment]
Ooh, this looks pretty major!  https://chapel-lang.org/perf/chapcs/?startdate=2024/08/01&enddate=2024/08/12&graphs=binarytreesshootoutbenchmarkn21,nbodyvariations,submittednbodyshootoutbenchmark

I haven't read the rules for this benchmark, or other implementations of it, in years.  Have you browsed enough to feel confident that this would be an acceptable entry?  Are you thinking of submitting it?

One minor change I'd make before submitting would be to change:

```chapel
   contributed by Casey Battaglino, Ben Harshbarger, Brad Chamberlain, and Jade Abraham
```
  into:
```chapel
   contributed by Jade Abraham
   based on the Chapel #3 version by Battaglino, Harshbarger, and Chamberlain
```

(or include our first names and have it take up two 80-column lines) to match what we've done for other cases.

**Comment by jabraham17** [comment]
> I haven't read the rules for this benchmark, or other implementations of it, in years. Have you browsed enough to feel confident that this would be an acceptable entry? Are you thinking of submitting it?

Yes, at the very least other entries do this as well. See https://benchmarksgame-team.pages.debian.net/benchmarksgame/program/binarytrees-gpp-5.html. I would love to get this submitted!

I will open a followup PR to make your suggested adjustments

**Pull request merged**